test/test_unit_strings.cpp

```diff
@@ -984,6 +988,23 @@ TEST(stringToUnits, invalid)
     EXPECT_FALSE(is_valid(unit_from_string("_liquid_()")));
 }
 
+TEST(stringToUnits, ParseIssues)
+{
+    auto u1 = unit_from_string("Metres");
+    EXPECT_EQ(u1, precise::m);
+
+    u1 = unit_from_string("degrees C");
+    EXPECT_EQ(u1, precise::degC);
+
+    u1 = unit_from_string("gramm");
+    EXPECT_EQ(u1, precise::g * precise::m);
+    u1 = unit_from_string("kilogramm");
+    EXPECT_EQ(u1, precise::kilo * precise::g * precise::m);
```

Comment:
I feel that this is a very surprising behavior. Mixing full words (kilo and gram) with symbols (m), especially without an operator or spacing in between is not really intuitive and of limited use? Supporting `gram*m` and `gram m` seems ok, but `gramm`?